Source/Diagnostics/ParticleIO.cpp

```diff
@@ -107,7 +109,85 @@ MultiParticleContainer::Restart (const std::string& dir)
     // - laser_names
     // we don't need to read back the laser particle charge/mass
     for (unsigned i = 0, n = species_names.size(); i < n; ++i) {
```

Comment:
After this function in a separate look, we should read back the laser particle positions.

They are stored in the 2nd part of `allcontainers`